config/class-sync.php

```diff
@@ -22,17 +32,27 @@ public static function instance() {
 	}
 
 	public function init() {
+		// Saving the initial blog_id in the init to be extra sure nobody changed it later on via a switch_to_blog().
+		$this->original_blog_id = get_current_blog_id();
```

Comment:
@WPprodigy I'm not sure if placing this here is early enough or if we should add it even before the init, to make sure we are getting the "first blog_id". Ideas?

Reply:
This should be good, happens on the `init` WP hook right now which is quite early. If we want to be really safe, could make sure it can't be re-set (in case some code triggers the hook again). So something like this perhaps would be good:

```
if ( is_null( $this->original_blog_id ) ) {
	$this->original_blog_id = get_current_blog_id();
}
```